src/main/java/ch/njol/skript/effects/EffReplace.java

```diff
@@ -140,11 +141,20 @@ private void replace(Event event, Object[] needles, Expression<?> haystackExpr)
 	
 	@Override
 	public String toString(@Nullable Event event, boolean debug) {
+		SyntaxStringBuilder builder = new SyntaxStringBuilder(event, debug);
+
+		builder.add("replace");
 		if (replaceFirst)
-			return "replace first " + needles.toString(event, debug) + " in " + haystack.toString(event, debug) + " with " + replacement.toString(event, debug)
-					+ "(case sensitive: " + caseSensitive + ")";
-		return "replace " + needles.toString(event, debug) + " in " + haystack.toString(event, debug) + " with " + replacement.toString(event, debug)
-				+ "(case sensitive: " + caseSensitive + ")";
+			builder.add("the first");
+		builder.add(needles);
+		builder.add("in");
+		builder.add(haystack);
+		builder.add("with");
+		builder.add(replacement);
```

Comment:
this might be a bit messy, but it'd be nice to just ``.add(needles, "in", haystack, "with", replacement)``
it's much harder to parse the actual string when it's vertical like this

Reply:
i like this, good idea

Reply:
although this could also just be done with multiple chained `append` calls (as append returns `this`), which would keep one append per expression. wdyt?

Reply:
that doesn't really solve the problem of breaking the phrase up into small segments that you then have to parse together